### DISCUSSION TITLE
Using Generic Dataset loader NaN issue

### DISCUSSION THREAD

**Discussion by Juliettengango1** [neuralhydrology/neuralhydrology#196] [Q&A]
Hello everyone:

I am using a generic dataset in my config file to run a NH model. The model runs well, but I am getting constant warning about my simulated values for some of the basins being invalid even by accessing these basins, I can see the valid values. Here are the warning I received:

2024-08-06 13:54:41,827: These basins do not have a single valid sample in the train period: ['1013', '1015', '1031', '1082', '1106', '1110', '1120', '1140', '1154', '1172', '1173', '1196', '1202', '1203', '1204', '1205', '1206', '1207', '1208', '1209', '1210', '1211', '1212', '1213', '1215', '1224', '1243', '1249', '1251', '1252', '1259', '1271', '1272', '1280', '308', '340', '353', '358', '369', '380', '386', '414', '416', '430', '4
31', '450', '466', '478', '515', '519', '544', '549', '561', '574', '586', '614', '626', '628', '629', '694', '698', '701', '709', '737', '740', '
745', '748', '757', '780', '788', '804', '823', '827', '837', '839', '841', '898', '904', '905', '940', '969', '990', '991', '998']
# Epoch 1: 100%|██████████| 1872/1872 [00:29<00:00, 62.96it/s, Loss: 0.0367]
2024-08-06 13:55:15,768: Epoch 1 average loss: avg_loss: 0.05920, avg_total_loss: 0.05920
Metrics for 1D are calculated over last 1 elements only. Ignoring 364 predictions per sequence.
# Validation:   1%|          | 8/804 [00:03<06:13,  2.13it/s]2024-08-06 13:55:19,996: Basin 1013 All simulated values are NaN, thus metrics will b
e NaN, too.
# Validation:   1%|          | 10/804 [00:04<07:35,  1.74it/s]2024-08-06 13:55:21,360: Basin 1015 All simulated values are NaN, thus metrics will 
be NaN, too.
# Validation:   2%|▏         | 14/804 [00:06<04:52,  2.71it/s]2024-08-06 13:55:22,796: Basin 1031 All simulated values are NaN, thus metrics will 
be NaN, too.
# Validation:   8%|▊         | 65/804 [00:25<05:07,  2.41it/s]2024-08-06 13:55:42,230: Basin 1110 All simulated values are NaN, thus metrics will 
be NaN, too.
# Validation:   9%|▉         | 71/804 [00:27<03:52,  3.15it/s]2024-08-06 13:55:44,348: Basin 1116 All simulated values are NaN, thus metrics will 
be NaN, too.

**Comment by danklotz** [comment]
Hello Juliette, 

you say that by accessing the basins you can see that the respective values are valid, even if the warnings tell you that the simulations are invalid (this often happens if you have NaNs in the inputs, but there can also be other reasons...). Is this correct? If yes, how did you check?

**Comment by Juliettengango1** [comment]
There were a few basins with missing values and I removed them, but the warnings persist. I used this command to check the netcdf files : "ncks --cln -v <var-name> <basin.nc>" I also tried to edit the metrics to use nanmean() instead of mean() to deal with the NaN issue, but that also didn't help. Would you then recommend to double check the data to remove all NaN?

**Reply by danklotz** [reply]
I would not necessarily to remove all NaN, but double checking the data is for sure a good idea. I am still not sure what you are checking for. Do you check that for the given basins all inputs are free of nans? I also assume that if all simulations of a given basin are nan, ten nanmean would not solve your problem. Right?
